### PR TITLE
CHE-567: add directive for formating output

### PR DESCRIPTION
**Pull request by ashumilova**
**_2 Upvotes**_

### CODE REVIEW

**Comment by ashumilova** [comment]
output

File: dashboard/src/components/attribute/format-output/che-format-output.directive.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      let lines = value.split('\n');
+      let result = '';
+
+      lines.forEach((line) => {
+
+        for (var i = 0; i < this.outputColors.length; i++ ) {
+          let outputColor = this.outputColors[i];
+          line = line.replace('[' + outputColor.type + ']', '[<span style=\"color: ' + outputColor.color + '\">' + outputColor.type + '</span>]');
+
+          //Additional formatting cause may have output like this - [INFO ]
```
Comment (benoitf) [inline dashboard/src/components/attribute/format-output/che-format-output.directive.js]: could we use a regexp to match`[<any space>type<any space>]`?

also replace by default works only on the first occurence
so I think we have [INFO] extracting [INFO] 10% only first occurence will be matched

File: dashboard/src/components/attribute/format-output/che-format-output.directive.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  /**
+   * Keep reference to the model controller
+   */
+  link($scope, element, attr) {
+    $scope.$watch(attr.ngModel, (value) => {
+      if (!value || value.length === 0) {
+        return;
+      }
+
+      let lines = value.split('\n');
```
Comment (benoitf) [inline dashboard/src/components/attribute/format-output/che-format-output.directive.js]: I think this code is a good candidate to have unit tests so we can check if with all our input values, the result is expected

File: dashboard/src/components/attribute/format-output/che-format-output.directive.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+          let outputColor = this.outputColors[i];
+          line = line.replace('[' + outputColor.type + ']', '[<span style=\"color: ' + outputColor.color + '\">' + outputColor.type + '</span>]');
+
+          //Additional formatting cause may have output like this - [INFO ]
+          if ('INFO' === outputColor.type) {
+            line = line.replace('[' + outputColor.type + ' ]', '[<span style=\"color: ' + outputColor.color + '\">' + outputColor.type + '</span>]');
+          }
+        }
+
+        result += '<br/>' + line;
```
Comment (benoitf) [inline dashboard/src/components/attribute/format-output/che-format-output.directive.js]: I think this code always add a blank carriage return at the beginning of the output while output should not have a "blank line" first

**Comment by benoitf** [comment]
+1

**Comment by olexii4** [comment]
+1

**Comment by riuvshin** [comment]
why this was merged? build fail with those  changes.

**Comment by garagatyi** [comment]
I supose that answer is because we don't have trigger that shows if PR brake the build

**Comment by ashumilova** [comment]
build failed cause PR contained adding new auto-generated file. it was not failing locally, cause that file was on file system

**Comment by benoitf** [comment]
yes it was issue only when we were running tests without having previously running app (which is only matching jenkins, not real life)
but yes as @garagatyi said, I would be a lot easier to have PR being checked automatically

**Pull request merged**